posthog/models/utils.py

```diff
@@ -280,3 +283,32 @@ def execute_with_timeout(timeout: int, database: str = "default") -> Iterator[Cu
         with connections[database].cursor() as cursor:
             cursor.execute("SET LOCAL statement_timeout = %s", [timeout])
             yield cursor
+
+
+def WARN_ON_CASCADE(collector, field, sub_objs, using):
+    """
+    Similar to ``django.db.models.CASCADE``, but intended for cases where we are
+    planning to change an existing ``CASCADE`` to a more restrictive replacement
+    and want to verify that doing so will not cause any issues before making
+    that change.
+
+    This also differs from the default ``CASCADE`` implementation in that it
+    does not try to get clever with nullable foreign keys -- this just treats
+    nullable foreign keys in the same way as non-nullable ones (i.e. it makes
+    sure to delete child instances _before_ parent objects, allowing them to be
+    referenced safely in post-deletion hooks.)
+    """
+    if sub_objs.exists():
+        logger.warn(
+            "cascading deletion of %r via %r is not side-effect free",
+            field.model,
+            field,
+            stack_info=True,
+        )
```

Comment:
If you run tests like this:

```
DJANGO_LOG_LEVEL=INFO py.test --log-cli-level warn posthog/api/test/test_person.py -k test_delete_person
```

You get to see these (with a big honkin' traceback):

```
WARNING  posthog.models.utils:utils.py:302 cascading deletion of <class 'posthog.models.person.person.PersonDistinctId'> via <django.db.models.fields.related.ForeignKey: person> is not side-effect free
```